analytics.py

```diff
@@ -0,0 +1,146 @@
+import math
+from assignment_06 import utils
+
+def check_significant(lower, upper, observed_avg):
+    significance = False
+    if(upper - lower <= observed_avg):
+        significance = True
+    return significance 
+
+def compute_critical(listOfAvgNNDistances):
+    criticalPoints = []
+    criticalPoints[0] = min(listOfAvgNNDistances)
+    criticalPoints[1] = max(listOfAvgNNDistances)
+    return criticalPoints
+
+def permutations(p):
+    listOfAvgNNDistances = []
+    
+    for x in range(p):
+        tmpList = []
+        tmpList.append(create_random_points(100))
+        listOfAvgNNDistances.append(average_nearest_neighbor_distance(tmpList))
+
+    return listOfAvgNNDistances    
+
+def average_nearest_neighbor_distance(points):
+    """
+    Given a set of points, compute the average nearest neighbor.
+    Parameters
+    ----------
+    points : list
+             A list of points in the form (x,y)
+    Returns
+    -------
+    mean_d : float
+             Average nearest neighbor distance
+    References
+    ----------
+    Clark and Evan (1954 Distance to Nearest Neighbor as a
+     Measure of Spatial Relationships in Populations. Ecology. 35(4)
+     p. 445-453.
+    """
+    listOfDistances = []
+    
+    for n in points:
+        lowest = 100
+        for x in points:
+            if euclidean_distance(n,x)!= 0 and euclidean_distance(n,x) < lowest:
+                lowest = euclidean_distance(n,x)
+        listOfDistances.append(lowest)
+
+    totalOfDistances = 0
+
+    for z in listOfDistances:
+        totalOfDistances += z
+
+    mean_d = totalOfDistances/len(points)
+
+    return mean_d
+    
+ def euclidean_distance(a, b):

```

Comment:
TravisCI is complaining about an indentation error here.  Indentation looks good though.  I see this sometimes if I use a text editor for a bit (with 4 spaces) and then switch to an IDE using tab.  Not sure if that is the case, and not a big deal.  More just an FYI incase you see that error in the future for your own work.